xonsh/history.py

```diff
@@ -290,6 +290,15 @@ def append(self, cmd):
         hf : HistoryFlusher or None
             The thread that was spawned to flush history
         """
+        opts = builtins.__xonsh_env__.get('HISTCONTROL', set())
```

Comment:
There is no need to provide a default value here as it will override the default value in `xonsh.environ.DEFAULT_VALUES`